core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -249,7 +249,14 @@ private[spark] class ExecutorAllocationManager(
    * yarn-client mode when AM re-registers after a failure.
    */
   def reset(): Unit = synchronized {
-    initializing = true
+    /**
+     * When some tasks need to be scheduled and initial executor = 0, resetting the initializing
+     * field may cause it to not be set to false in yarn.
+     * SPARK-20079: https://issues.apache.org/jira/browse/SPARK-20079
+     */
+    if (maxNumExecutorsNeeded() == 0) {
+      initializing = true
```

Comment:
For the 2nd point `an executor idle timeout has elapsed.`, in the `reset` we also clear the executor idle timeout tracking info, which means executor idle timeout will be recalculated based on newly registered executors. I think this point could be true.

Yes we could set this flag `initializing` to false in reset, but I think the scenario I mentioned above still should be concerned.

>the purpose of "initializing" is to avoid unnecessary executors ramp down before the stage submitted or executor timeout. For example if min executor number is 0, initial number is 10. If "initializing" is set to false, executor number will ramp down to 0 immediately, and during this time if stage is submitted, then still requires unnecessary executor ramp up to meet this stage's requirement.